compiler/rustc_const_eval/src/interpret/place.rs

```diff
@@ -1018,31 +1018,53 @@ where
         self.allocate_dyn(layout, kind, MemPlaceMeta::None)
     }
 
-    /// Returns a wide MPlace of type `str` to a new 1-aligned allocation.
-    /// Immutable strings are deduplicated and stored in global memory.
-    pub fn allocate_str(
+    /// Allocates a sequence of bytes in the interpreter's memory.
+    /// For immutable allocations, uses deduplication to reuse existing memory.
+    /// For mutable allocations, creates a new unique allocation.
+    pub fn allocate_bytes(
         &mut self,
-        str: &str,
+        bytes: &[u8],
+        align: Align,
         kind: MemoryKind<M::MemoryKind>,
         mutbl: Mutability,
     ) -> InterpResult<'tcx, MPlaceTy<'tcx, M::Provenance>> {
-        let tcx = self.tcx.tcx;
-
-        // Use cache for immutable strings.
         let ptr = if mutbl.is_not() {
-            // Use dedup'd allocation function.
+            // Deduplicate immutable allocations using a salted hash
```

Comment:
You have changed all the comments in this function. The new comments aren't wrong, but they are also not clearly better than the old ones. Why have you rewritten them?
The salted hash part isn't relevant here, what is relevant is that we use the `dedup` function, so for this line here please restore the previous comment.

When your comment is a sentence, please make sure to end it in `.`.

Reply:
Ah, Apologies for my earlier misunderstanding based on the function name. I’ve restored the original comments as per your feedback.